internal/trie/node/hash_test.go

```diff
@@ -241,7 +241,7 @@ func Test_Leaf_EncodeAndHash(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			t.Parallel()
 
-			encoding, hash, err := testCase.leaf.EncodeAndHash()
+			encoding, hash, err := testCase.leaf.EncodeAndHash(false)
```

Comment:
please add `isRoot` as a field to the test case. And also add a test case with `isRoot: true`.